packages/turbo-repository/rust/src/lib.rs

```diff
@@ -291,4 +291,39 @@ impl Workspace {
 
         Ok(serializable_packages)
     }
+
+    // Given a path (relative to the workspace root), returns the
+    // package that contains it.
+    //
+    // This is a naive implementation that simply "iterates-up". If this function is
+    // expected to be called many times for files that are deep within the same
+    // package, we could optimize this by caching the containing-package of
+    // every ancestor.
+    #[napi]
+    pub async fn find_package_by_path(&self, path: String) -> Result<Package, Error> {
+        let package_mapper = DefaultPackageChangeMapper::new(&self.graph);
+        let anchored_path = AnchoredSystemPath::new(&path)
+            .map_err(|e| Error::from_reason(e.to_string()))?
+            .clean();
+        match package_mapper.detect_package(&anchored_path) {
+            turborepo_repository::change_mapper::PackageMapping::All(
+                _all_package_change_reason,
+            ) => Err(Error::from_reason("file belongs to many packages")),
```

Comment:
An alternative here would be to possibly return the root package or return that the file change is considered "global".